cmd/qvain-backend/api_proxy.go

```diff
@@ -223,43 +222,43 @@ func (api *ApiProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	// only allow access to /directories/ and /files/; path has been cleaned by Go on instantiation
 	if !(strings.HasPrefix(r.URL.Path, "/directories/") || strings.HasPrefix(r.URL.Path, "/files/")) {
-		jsonError(w, "access denied", http.StatusForbidden)
+		loggedJSONError(w, "access denied", http.StatusForbidden, &api.logger).Msg("error in api_proxy")
 	}
 
 	// make sure the user is authenticated
 	session, err := api.sessions.UserSessionFromRequest(r)
 	if err != nil {
-		sessionError(w, err)
+		sessionError(w, err, &api.logger).Err(err).Msg("Error: User is not authenticated")
 		return
 	}
 
 	// allowed_projects should be set by the proxy, not in the original request
 	query := r.URL.Query()
 	if _, found := query["allowed_projects"]; found {
-		jsonError(w, "bad request: allowed_projects is not allowed", http.StatusBadRequest)
+		loggedJSONError(w, "bad request: allowed_projects is not allowed", http.StatusBadRequest, &api.logger).Msg("error in api_proxy")
 		return
 	}
 
 	// proxy takes care of converting project to project_identifier as needed
 	if _, found := query["project_identifier"]; found {
-		jsonError(w, "bad request: project_identifier is not allowed", http.StatusBadRequest)
+		loggedJSONError(w, "bad request: project_identifier is not allowed", http.StatusBadRequest, &api.logger).Msg("error in api_proxy")
 		return
 	}
 
 	// check optional project query parameter
 	if projectQueries, found := query["project"]; found {
 		if len(projectQueries) > 1 {
-			jsonError(w, "bad request: multiple projects in query", http.StatusBadRequest)
+			loggedJSONError(w, "bad request: multiple projects in query", http.StatusBadRequest, &api.logger).Msg("error in api_proxy")
 			return
 		}
 		if len(session.User.Projects) < 1 {
-			jsonError(w, "access denied: user has no projects", http.StatusForbidden)
+			loggedJSONError(w, "access denied: user has no projects", http.StatusForbidden, &api.logger).Msg("error in api_proxy")
 			return
 		}
 		project := projectQueries[0]
 		if !session.User.HasProject(project) {
 			api.logger.Debug().Strs("projects", session.User.Projects).Str("wanted", project).Msg("project check")
-			jsonError(w, "access denied: invalid project", http.StatusForbidden)
+			loggedJSONError(w, "access denied: invalid project", http.StatusForbidden, &api.logger).Msg("error in api_proxy")
```

Comment:
invalid error message..

Reply:
I wonder if we need to have this "error in api_proxy" at all, what if we would just enable the "caller" field in the logs?

https://github.com/rs/zerolog/commit/27e0a22cbc6f6a8f2a7ac2ed2c9b410b05f59e02